libqtile/backend/wayland/core.py

```diff
@@ -253,6 +242,30 @@ def _on_cursor_motion_absolute(self, _listener, event: pointer.PointerEventMotio
     def _on_new_virtual_keyboard(self, _listener, virtual_keyboard: VirtualKeyboardV1):
         self._add_new_keyboard(virtual_keyboard.input_device)
 
+    def _on_new_layer_surface(self, _listener, layer_surface: layer_shell_v1.LayerSurfaceV1):
+        logger.debug("Signal: layer_shell new_surface_event")
+        assert self.qtile is not None
+
+        wid = self._get_new_wid()
+        win = window.Static(self, self.qtile, layer_surface, wid)
+        logger.info(f"Managing new layer_shell window with window ID: {wid}")
+        self.qtile.manage(win)
+
+    def _on_new_toplevel_decoration(
+        self, _listener, decoration: xdg_decoration_v1.XdgToplevelDecorationV1
+    ):
+        logger.debug("Signal: xdg_decoration new_top_level_decoration")
+        decoration.set_mode(xdg_decoration_v1.XdgToplevelDecorationV1Mode.SERVER_SIDE)
+
+    def _get_new_wid(self):
+        wid = 0
+        wids = self.qtile.windows_map.keys()
+        while True:
+            if wid not in wids:
+                break
+            wid += 1
+        return wid
```

Comment:
Why not just `max(self.qtile.windows_map.keys()) + 1`?

Reply:
Yeah that'd work. Could any problem arise by having an ever-increasing number?

Reply:
I think the performance problem you'll get by iterating through all windows ids before creating a new window will hit you before any problem with a number that's too big, but I'm just guessing.

Reply:
You don't need to have `INT_MAX` windows existing to have a window with ID `INT_MAX`. With the current version, you can have two windows with IDs 0 and 2, and the next window will get ID 1. But the every-increasing version can do:

1. have window with ID 0
2. next window gets ID 1
3. close window 0
4. next window gets ID 2
5. next window only needs to iterate through 2 windows

You could get a too big number with never opening more than 2 windows. Unless I'm misunderstanding what you meant?

I'm more wondering out of curiousity... tbh I would assume it would take literally years to get an error by incrementing the int too much :stuck_out_tongue:

Then again, doesn't `max(self.qtile.windows_map.keys()) + 1` iterate through them all anyway?

Reply:
OK im gonna go with your version, it makes the code easier to understand and the difference either way is gonna be like microseconds or something lol

Reply:
> Then again, doesn't max(self.qtile.windows_map.keys()) + 1 iterate through them all anyway?

Yes but it only does it once. With `if wid not in wids`, you're doing it N times here, where N is the number of attempts at brute-forcing the first available wid. Not exactly the same time complexity. :)

> You don't need to have INT_MAX windows existing to have a window with ID INT_MAX.

No need to explain, I had it already. :) 

> the difference either way is gonna be like microseconds or something lol

Yeah I think we're miles ahead any problem both ways.